dio/lib/src/breadcrumb_client_adapter.dart

```diff
@@ -0,0 +1,71 @@
+// ignore_for_file: strict_raw_type
+
+import 'dart:typed_data';
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+
+import 'adapter/dio_adapter.dart';
+
+/// A [Dio](https://pub.dev/packages/dio)-package compatible HTTP client adapter
+/// which records requests as breadcrumbs.
+///
+/// Remarks:
+/// If this client is used as a wrapper, a call to close also closes the
+/// given client.
+class BreadcrumbClientAdapter extends HttpClientAdapter {
+  // ignore: public_member_api_docs
+  BreadcrumbClientAdapter({HttpClientAdapter? client, Hub? hub})
+      : _hub = hub ?? HubAdapter(),
+        _client = client ?? createAdapter();
```

Comment:
What about not using a default adapter here and making the client required. That would avoid all the dio export mess.

Reply:
That would just move the burden of creating this code to the user :(

Or could the iser do something like:
```dart
final dio = Dio();
dio.httpClientAdapter = SentryHttpClientAdapter(client: dio.httpClientAdapter);
```

I guess that would be acceptable.

Reply:
Changed it